xchange-core/src/main/java/org/knowm/xchange/exceptions/SystemOverloadException.java

```diff
@@ -0,0 +1,21 @@
+package org.knowm.xchange.exceptions;
+
+/** An exception indicating there the rate limit for making requests has been exceeded */
```

Comment:
then is the system overloaded in general (as the name of the exception suggests) or is the call rate for your account exceeded (as the comment suggests)

for the former we already have an exception type: RateLimitExceededException

Reply:
My idea is to manage overload in general i.e. when server return an http error 502.
I added the particular exception because with the latest changes Bitmex client generates a generic exception when the server returns the error in attached image; then my software cannot understand what is really happened. Before introduction of "updateRateLimit" function, Bitmex client didn't trap the exception and I could manage 502 error;  if you have a better idea to return the specific error, I can modify my code and create a pull request.

![image](https://user-images.githubusercontent.com/22959765/50488414-bbe3ed80-0a02-11e9-8315-44a110bdaeb4.png)

Reply:
still the name of your class says one thing and the comment above it says another, fix the comment then

Reply:
look ok to me now